CHANGELOG/v1.10.0-rc.0.md

```diff
@@ -0,0 +1,417 @@
+🚨 This is a RELEASE CANDIDATE. Use it only for testing purposes. If you find any bugs, file an [issue](https://github.com/kubernetes-sigs/cluster-api/issues/new).
+
```

Comment:
I think it would be good to have the highlights + Deprecation and Removals Warning like in previous release notes

Reply:
Added